crates/proto/src/xfer/dnssec_dns_handle.rs

```diff
@@ -420,33 +420,24 @@ where
 
     // check the DNSKEYS against the trust_anchor, if it's approved allow it.
     //   this includes the root keys
-    {
-        let anchored_keys: Vec<&DNSKEY> = rrset
-            .records()
-            .iter()
-            .map(|r| r.data())
-            .filter_map(DNSKEY::try_borrow)
-            .filter(|dnskey| {
-                if handle
-                    .trust_anchor
-                    .contains_dnskey_bytes(dnskey.public_key())
-                {
-                    debug!(
-                        "validated dnskey with trust_anchor: {}, {}",
-                        rrset.name(),
-                        dnskey
-                    );
-
-                    true
-                } else {
-                    false
-                }
-            })
-            .collect::<Vec<_>>();
+    for r in rrset.records().iter() {
+        let Some(key_rdata) = DNSKEY::try_borrow(r.data()) else {
+            continue;
+        };
 
-        if !anchored_keys.is_empty() {
-            return Ok(true);
+        if !handle
```

Comment:
nit: I think flipping the condition makes this shorter
``` rust
if is_trust_anchor() {
    debug!("something");
    return Ok(true)
}
```
the `continue` is a bit redundant since the loop will continue at the end of the scope

Reply:
This is true but I'm optimizing for maintainability here: if we want to add more conditions that trigger non-validation, I'd rather add them as early continues than the other way around.